Add application for audio retargeting

GiulioRomualdi opened the pull request:
cc @S-Dafarra

S-Dafarra reviewed (approved):
Great!

kouroshD commented on app/scripts/audioForRetargeting.xml:
```diff
@@ -0,0 +1,52 @@
+  </module>
+
+  <!-- Connections -->
+  <!-- Hand retargeting  -->
```
Probably this is a leftover, you can delete this.

kouroshD commented:
@GiulioRomualdi Thank you for the PR, that's great. Before merging, can you please link to the issue here. It is not clear the functionalities of different options in yarp devices and the functionalities of the modules; or you can add the explanation here.

GiulioRomualdi commented:
Following @randaz81 suggestions https://github.com/robotology/yarp/issues/2388#issuecomment-713418289 I dropped `yarphear` in favor of `yarpdev --device AudioPlayerWrapper --subdevice portaudioPlayer`.

I've also added a call to the `rpc` port in the `xml` file. Thanks to this we can avoid to open a new terminal to `start` the process. 

In order to use the application we just have to `run all` e then `connect all`

S-Dafarra commented:
@kouroshD @GiulioRomualdi can we merge this?

GiulioRomualdi commented:
From my side yes

kouroshD commented:
@GiulioRomualdi  @S-Dafarra  Please add a documentation to the xml file. At least wringing the functionality of each module, i.e., `AudioRecorderWrapper` and `AudioPlayerWrapper`. If you can write about the parameters it will be great.

GiulioRomualdi commented:
I don't have any idea of the meaning of some parameters passed to the `yarpdev`. I've just finished the work done by @miladshafiee in the past. 
However, I'm pretty sure that:
- `AudioRecorderWrapper`: it's a device required to record the audio from the microphone
- `--name <port_name>`: the options `--name` is used to set the name of the port
- `AudioPlayerWrapper` it's a device required to hear the audio.

The ports opened by `AudioRecorderWrapper` are connected to the one opened by `AudioPlayerWrapper`

kouroshD commented:
@GiulioRomualdi  Thank you for the explanation, we can go on with the merge.
